Public/Src/Cache/ContentStore/Library/FileSystem/PassThroughFileSystem.cs

```diff
@@ -445,89 +446,86 @@ private async Task CopyFileInsideSemaphoreAsync(AbsolutePath sourcePath, Absolut
             }
         }
 
-        private Stream OpenInternal(AbsolutePath path, FileAccess accessMode, FileMode mode, FileShare share, FileOptions options, int bufferSize)
+        private Stream TryOpenFile(AbsolutePath path, FileAccess accessMode, FileMode mode, FileShare share, FileOptions options, int bufferSize)
         {
-                if (OperatingSystemHelper.IsUnixOS)
-                {
-                    if (DirectoryExists(path))
-                    {
-                        throw new UnauthorizedAccessException($"Cannot open directory {path} as a file.");
-                    }
+            if (OperatingSystemHelper.IsUnixOS)
+            {
+                return TryOpenFileUnix(path, accessMode, mode, share, options, bufferSize);
+            }
 
-                    if (mode == FileMode.Open && !FileExists(path))
-                    {
-                        return null;
-                    }
+            return TryOpenFileWin(path, accessMode, mode, share, options, bufferSize);

```

Comment:
Very minor: I'm not opposed to early returns, but for two equal options of equal complexity, I think having an else block is more natural.